src/IB/IBFEMethod.cpp

```diff
@@ -521,6 +547,30 @@ IBFEMethod::preprocessIntegrateData(double current_time, double new_time, int nu
 
     d_started_time_integration = true;
 
+    // Determine if we need to execute reinitElementMappings().
+    bool do_reinit_element_mappings = false;
+    assertStructureOnFinestLevel();
+    const double patch_dx = get_min_patch_dx(d_hierarchy, d_hierarchy->getFinestLevelNumber());
+    for (unsigned int part = 0; part < d_meshes.size(); ++part)
+    {
+        // To avoid allocating an extra vector we abuse "new" for a moment since
+        // it gets overwritten below anyway
+        PetscVector<double>& diff = d_X_vecs->get("new", part);
+        PetscVector<double>& last = d_X_vecs->get("last_patch_elem_assoc", part);
+        VecWAXPY(diff.vec(), -1.0, last.vec(), d_X_vecs->get("solution", part).vec());
+        const double max_distance = diff.linfty_norm();
```

Comment:
Should we look at the pointwise Euclidean norm of the displacement?

Reply:
That's a good question. This l-oo (that's a lower-case L) type norm makes more sense to me because a particle could move 0.75 cell widths in each coordinate direction and still be inside the same cell (and here that is reflected by having a maximum distance of 0.75) but if we use the L2 norm then we would get `sqrt(3*0.75**2) = 1.299...`, implying that we have moved outside the cell.

Reply:
I'm not sure what the right choice is - this is an odd CFL condition because the CFL condition is inherently Eulerian but we have a Lagrangian description of the structure.

Reply:
I think that we also should consider the motion of the edges/faces. I think we want to think about the furthest that any material point in the element could move. Specifically, we need to detect if is it possible that a material point in the element that was not previously able to interact with the Cartesian grid patch is now able to do so. It may be best to use the max norm in making this determination. I need to draw a picture, but I don’t expect that I will get a chance until this afternoon.

Reply:
Independent of the norm choice, we do have some wiggle room here - we associate elements with patches if they are within 1 cell width of the patch but we reassociate (in this case, with the default values) once a point has move 0.75 cell widths in a coordinate direction.

Reply:
One more useful point: none of the tests, aside from the new one, actually execute this reassociation procedure because the CFL based regridding is much more conservative.

Reply:
Sure, I understand --- I just want to try to make sure that we don't think we have more wiggle room than we actually do and get surprised later on. :-)